window-copy.c

```diff
@@ -3048,31 +3066,50 @@ window_copy_search_jump(struct window_mode_entry *wme, struct grid *gd,
 		return (window_copy_search_jump(wme, gd, sgd,
 		    direction ? 0 : gd->sx - 1,
 		    direction ? 0 : gd->hsize + gd->sy - 1, fy, cis, 0,
-		    direction, regex, foundlen));
+		    direction, regex));
 	}
 	return (0);
 }
 
+static void
+window_copy_move_after_search_mark(struct window_copy_mode_data *data,
+    u_int *fx, u_int *fy, int wrapflag)
+{
+	struct screen  *s = data->backing;
+	u_int		at, start;
+
+	if (window_copy_search_mark_at(data, *fx, *fy, &start) == 0 &&
+	    data->searchmark[start] != 0) {
+		while (window_copy_search_mark_at(data, *fx, *fy, &at) == 0) {
+			if (data->searchmark[at] != data->searchmark[start])
```

Comment:
We should also break if `wrapflag` is 0 and we are at the end of the grid. Otherwise this will cause an infinite loop. This can be reproduced by starting a program which displays a string at the bottom edge of the pane (such as another tmux) and searching for that string.

Reply:
Ah, how do I detect the end of the grid?

Reply:
Same as `window_copy_move_right`: `fx == gd->sx - 1; fy == gd->hsize + gd->sy - 1`.

Reply:
Amended dd4796d

Reply:
This is now fixed, thanks!